runtime/src/main/java/com/flipkart/flux/deploymentunit/ExecutableRegistryPopulator.java

```diff
@@ -0,0 +1,146 @@
+
+                //create a class loader and get required classes from it
+                URLClassLoader classLoader = deploymentUnitUtil.getClassLoader(deploymentUnitName);
+                Class TaskClass = classLoader.loadClass(Task.class.getCanonicalName());
+                Class InjectorClass = loadClassLoaderInjector(classLoader);
+                Method getClassInstanceMethod = InjectorClass.getMethod("getClassInstance", Class.class);
+                Class guiceModuleClass = classLoader.loadClass("com.google.inject.Module");
+
+                Object injectorClassInstance;
+                String DUModuleClassFQN = String.valueOf(deploymentUnitUtil.getProperties(classLoader).getProperty("guiceModuleClass"));
+
+                //check if user has specified any guice module class name in deployment unit configuration file, if not create an empty injector
+                if(DUModuleClassFQN == null || DUModuleClassFQN.trim().isEmpty() || DUModuleClassFQN.equals("null"))
+                    injectorClassInstance = InjectorClass.newInstance();
+                else {
+                    injectorClassInstance = InjectorClass.getConstructor(guiceModuleClass).newInstance(classLoader.loadClass(DUModuleClassFQN).newInstance());
+                }
+
+                Set<Method> taskMethods = deploymentUnitUtil.getTaskMethods(classLoader);
+                //for every task method found in the deployment unit create an executable and keep it in executable registry
+                for(Method method : taskMethods) {
+                    String taskIdentifier = new MethodId(method).toString();
+                    Annotation taskAnnotation = method.getAnnotationsByType(TaskClass)[0];
+                    Class<? extends Annotation> annotationType = taskAnnotation.annotationType();
+                    long timeout = 1000l; //default timeout
+                    for (Method annotationMethod : annotationType.getDeclaredMethods()) {
+                        Object value = annotationMethod.invoke(taskAnnotation, (Object[])null);
+                        if(annotationMethod.getName().equals("timeout")) { //todo: find a way get Task.timeout() name
+                            timeout = (Long) value;
+                        }
+                    }
+
+                    Object singletonMethodOwner = getClassInstanceMethod.invoke(injectorClassInstance, method.getDeclaringClass());
+                    executableRegistry.registerTask(taskIdentifier, new ExecutableImpl(singletonMethodOwner, method, timeout, classLoader));
+
+                    //add the router name to deployment unit routers
+                    routerNames.add(method.getDeclaringClass().getName() + "_" + method.getName()); //convention: task router name is classFQN_taskMethodName
+                }
+            } catch (Exception e) {
+                LOGGER.error("Unable to populate Executable Registry for deployment unit: "+deploymentUnitName+". Exception: "+e.getMessage());
```

Comment:
For all logger usages, don't use string concatenation yourself. 
Use markup. Above would be Logger.error("Unable to.. unit: {}. Exception : ", deploymentUnitName,e);